### PR TITLE
405 personalized belt stvi

### PR DESCRIPTION
**Pull request by stanvins**

### CODE REVIEW

**Comment by stanvins** [comment]
Updated to work with the changed functions (especially fetchBooks). Now with weighted tags and so far limiting the tag array to the 10 heaviest.

File: src/client/redux/interaction.reducer.js [hunk 1]
```diff
@@ -11,13 +11,13 @@ const interactionReducer = (state = defaultState, action) => {
...
       const newState = Object.assign({}, state);
-      if (state.interactions.length > 0) {
-        newState.interactions.push({
-          type: 'INTERACTION',
-          interaction: action.interaction,
-          pid: action.pid
-        });
-      }
+
+      newState.interactions.push({
```
Comment (kfudbc) [inline src/client/redux/interaction.reducer.js]: Der er et lille problem med interactions array i forhold til den måde man skal opdatere state på "immutable" i redux. Derfor skal interactions også være en ny liste, da det "gamle state" ikke må mutere.

File: src/client/components/belt/RecommendationsBelt.container.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  if (cnt > 0 && current > 0) {
+    weightedArray.push({id: current, weight: cnt});
+  }
+
+  // sorting the array by heaviest weight
+  let sortedArray = sortByKey(weightedArray, 'weight').reverse();
+
+  // keeping only the top 10 tags
+  while (sortedArray.length > 10) {
+    sortedArray.pop();
```
[the remark this answers is not in the document]
Comment (stanvins) [reply src/client/components/belt/RecommendationsBelt.container.js]: Lodash is great. I am now doing it all with Lodash and as a bonus, the slice method appears to be working as it should.

File: src/client/redux/interaction.reducer.js (hunk 1 above)
Comment (stanvins) [reply src/client/redux/interaction.reducer.js]: Yes. I see that. I think I have fixed it now. Please check.

**Pull request merged**